File: src/main/java/me/rsman/BetterMinecraftCore/Managers/DamageManager.java
package me.rsman.BetterMinecraftCore.Managers;

import me.rsman.BetterMinecraftCore.BetterMinecraftCore;
import org.bukkit.entity.Player;

import java.util.Map;

public final class DamageManager {
    public static double getDamageFromAll(Player player){
        Map<String, Long> attributes = PlayerManager.getAttributes(player.getUniqueId().toString());
        double damage = Math.max((double)attributes.get("damage"),1);

        //apply strength
        damage *= (1 + (double)attributes.get("strength") / 100);

        //apply crit chance / damage
        if(Math.random()*100 < (double) attributes.get("critChance")){
            damage *= (1 + (double)attributes.get("critDamage") / 100);
        }

        return damage;
    }
}

File: src/main/kotlin/me/rsman/BetterMinecraftCore/Managers/DamageManager.kt
package me.rsman.BetterMinecraftCore.Managers

import org.bukkit.entity.Player

object DamageManager {
    fun getDamageFromAll(player: Player): Double {
        val attributes = PlayerManager.getAttributes(player.uniqueId.toString())
        var damage = (attributes["damage"]?.toDouble() ?: 1.0).coerceAtLeast(1.0)

        //apply strength
        damage *= 1 + (attributes["strength"]?.toDouble() ?: 0.0) / 100

        //apply crit chance / damage
        if (Math.random() * 100 < (attributes["critChance"]?.toDouble() ?: 0.0)) {
            damage *= 1 + (attributes["critDamage"]?.toDouble() ?: 0.0) / 100
        }
        return damage
    }
}